src/main/kotlin/snyk/container/ContainerService.kt

```diff
@@ -0,0 +1,139 @@
+package snyk.container
+
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import com.google.gson.reflect.TypeToken
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.project.Project
+import io.snyk.plugin.cli.CliError
+import io.snyk.plugin.cli.ConsoleCommandRunner
+import io.snyk.plugin.services.CliAdapter
+import org.jetbrains.annotations.TestOnly
+import snyk.common.SnykError
+import java.lang.reflect.Type
+import java.util.Collections
+
+private val LOG = logger<ContainerService>()
+
+@Service
+class ContainerService(project: Project) : CliAdapter<ContainerResult>(
+    project = project
+) {
+
+    private var imageCache: KubernetesImageCache = project.service()
+
+    @TestOnly
+    fun setKubernetesImageCache(cache: KubernetesImageCache) {
+        imageCache = cache
+    }
+
+    fun scan(): ContainerResult {
+        val images = imageCache.getKubernetesWorkloadImages()
+        return this.scan(images)
+    }
+
+    private fun scan(images: Set<KubernetesWorkloadImage>): ContainerResult {
+        LOG.debug("starting scanning container service")
+
+        // we want this synchronized so we can parallel process the stream that fills it
+        val containerIssueImageList = Collections.synchronizedList(mutableListOf<ContainerIssuesForImage>())
+
+        var commands = buildCliCommandsList(listOf("container", "test", "--app-vulns"))
+        val mutCommands = commands.toMutableList()
+        images.parallelStream().forEach { image ->
+            mutCommands.add(image.image)
+            commands = mutCommands.toList()
+            val result = execute(commands)
```

Comment:
what if we'll have 1000 images to test? will it be ok to run 1000 CLI instances in parralell?...